src/Tools/dotnet-monitor/CollectionRules/Actions/LoadProfilerAction.cs

```diff
@@ -0,0 +1,115 @@
+
+        public LoadProfilerActionFactory(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+        }
+
+        public ICollectionRuleAction Create(IEndpointInfo endpointInfo, LoadProfilerOptions options)
+        {
+            if (null == options)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            ValidationContext context = new(options, _serviceProvider, items: null);
+            Validator.ValidateObject(options, context, validateAllProperties: true);
+
+            return new LoadProfilerAction(_serviceProvider, endpointInfo, options);
+        }
+
+        internal sealed partial class LoadProfilerAction :
+            CollectionRuleActionBase<LoadProfilerOptions>
+        {
+            private readonly IEndpointInfo _endpointInfo;
+            private readonly LoadProfilerOptions _options;
+
+            public override CollectionRuleActionExecutionType ExecutionType
+            {
+                get => CollectionRuleActionExecutionType.AllowAsync;
+            }
+
+            public LoadProfilerAction(IServiceProvider serviceProvider, IEndpointInfo endpointInfo, LoadProfilerOptions options)
+                : base(endpointInfo, options)
+            {
+                _endpointInfo = endpointInfo ?? throw new ArgumentNullException(nameof(endpointInfo));
+                _options = options ?? throw new ArgumentNullException(nameof(options));
+
+                if (null == _options.Path || _options.PathMap.Count == 0)
+                {
+                    throw new ArgumentException(nameof(options.Path));
+                }
+
+                if (Guid.Empty == _options.ProfilerGuid)
+                {
+                    throw new ArgumentException(nameof(options.ProfilerGuid));
+                }
+            }
+
+            protected override async Task<CollectionRuleActionResult> ExecuteCoreAsync(
+                TaskCompletionSource<object> startCompletionSource,
+                CancellationToken token)
+            {
+                try
+                {
+                    DiagnosticsClient client = new DiagnosticsClient(_endpointInfo.Endpoint);
+
+                    ProcessInfo procInfo = await client.GetProcessInfoAsync(token);
+                    RidResolver ridResolver = new(procInfo);
+                    string choosenRid = ridResolver.Resolve(_options.PathMap.Keys);
+
+                    await client.SetStartupProfilerAsync(_options.ProfilerGuid, _options.PathMap[choosenRid], token);
+
+                    if (!startCompletionSource.TrySetResult(null))
+                    {
+                        throw new InvalidOperationException();
+                    }
+
+                    return new CollectionRuleActionResult()
+                    {
+                        OutputValues = new Dictionary<string, string>(StringComparer.Ordinal)
```

Comment:
I'm not certain either of these pieces of information are relevant outside of this action. Maybe for logging purposes, but then the action should just import an ILogger and log that information rather than surfacing it to the action execution system.